src/index.js

```diff
@@ -0,0 +1,284 @@
+import StatusApi from './statusApi.js';
+import {buildUrl} from './utils';
+const Ip = require('ip');
+const fs = require('fs');
+const path = require('path');
+
+// Duration (in ms) to wait before attempting to reconnect.
+const CONNECT_INTERVAL = 4000;
+const CHAIN_NAME = 'embark';
+const NETWORK_NAME = 'Embark';
+
+// private method symbols
+const _connectStatus = Symbol('connectStatus');
+const _connectToStatusNetwork = Symbol('connectToStatusNetwork');
+const _addAndConnectStatusNetwork = Symbol('addAndConnectStatusNetwork');
+const _getStatusNetworkId = Symbol('getStatusNetworkId');
+const _setStatusNetworkId = Symbol('setStatusNetworkId');
+const _registerEvents = Symbol('registerEvents');
+
+/**
+ * Plugin that connects an Embark dApp to the Status app, and allows the dApp
+ * to be run in the Status browser.
+ */
+class EmbarkStatusPlugin {
+  constructor(embark) {
+    this.networkId = 1337;
+    this.embark = embark;
+    this.events = this.embark.events;
+    this.pluginConfig = this.embark.pluginConfig;
+    this.deviceIp = this.pluginConfig.deviceIp;
+    this.logger = this.embark.logger;
+    // we want a timeout to occur before issuing the next request, so reduce the timeout specified
+    this.statusApi = new StatusApi({ip: this.deviceIp, logger: this.logger, timeout: CONNECT_INTERVAL - 1000});
+    this.webServerConfig = {};
+    this.blockchainConfig = {};
+    this.networkIdDirectory = path.join(process.env.DAPP_PATH, '.embark', 'embark-status');
+    this.networkIdPath = path.join(this.networkIdDirectory, 'networkId');
+
+    // gets hydrated webserver config from embark
+    this.events.on('config:load:webserver', webServerConfig => {
+      this.webServerConfig = webServerConfig;
+    });
+
+    // gets hydrated blockchain config from embark
+    this.events.on('config:load:blockchain', blockchainConfig => {
+      this.blockchainConfig = blockchainConfig;
+    });
+
+    // adds cors to blockchain and storage clients
+    this.events.request('config:cors:add', `http://${this.deviceIp}`);
+
+    // register service check
+    this[_registerEvents]();
+
+    // kick off network connection, and open dapp after connection
+    this.connectStatusToEmbark();
+  }
+
+  /**
+   * Continuously attempts to connect to the Status app by instructing
+   * the Status app to connect to this dApp's network (node) until a 
+   * successful connection is made. If the network doesn't exist, it is
+   * added. Once a connection is made, the Status app is instructed to
+   * open this dApp in it's dApp browser.
+   * 
+   * @returns {void}
+   */
+  connectStatusToEmbark() {
+    // keep trying to connect to the status app if previous connections were unsuccessful
+    this.events.once('web3Ready', (_isReady) => {
```

Comment:
Like I said on Slack, I renamed the event `blockchain:ready` to fit the new standard and because it's more intuitive.